samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js

```diff
@@ -1,45 +1,139 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-const { TeamsActivityHandler, CardFactory } = require('botbuilder');
+const { TeamsActivityHandler, CardFactory, TeamsInfo } = require('botbuilder');
 
 class TeamsMessagingExtensionsActionBot extends TeamsActivityHandler {
     handleTeamsMessagingExtensionSubmitAction(context, action) {
         switch (action.commandId) {
-        case 'createCard':
-            return createCardCommand(context, action);
-        case 'shareMessage':
-            return shareMessageCommand(context, action);
-        default:
-            throw new Error('NotImplemented');
+            case 'createCard':
+                return createCardCommand(context, action);
+            case 'shareMessage':
+                return shareMessageCommand(context, action);
+            default:
+                throw new Error('NotImplemented');
         }
     }
+
+    async handleTeamsMessagingExtensionFetchTask(context, action) {
+        var member;
+        try {
+            member = await this.getSingleMember(context);
+        } catch (e) {
+            if (e.code === 'BotNotInConversationRoster') {
+                return {
+                    task: {
+                        type: 'continue',
+                        value: {
+                            card: GetJustInTimeCardAttachment(),
+                            height: 400,
+                            title: 'Adaptive Card - App Installation',
+                            width: 300
+                        },
+                    },
+                };
+            }
+            throw e;
+        }
+        return {
+            task: {
+                type: 'continue',
+                value: {
+                    card: GetAdaptiveCardAttachment(),
+                    height: 400,
+                    title: 'Hello ' + member,
+                    width: 300
+                },
+            },
+        };
+    }
```

Comment:
```suggestion
	async handleTeamsMessagingExtensionFetchTask(context, action) {
        try {
            const member = await this.getSingleMember(context);
            return {
                task: {
                    type: 'continue',
                    value: {
                        card: GetAdaptiveCardAttachment(),
                        height: 400,
                        title: 'Hello ' + member,
                        width: 300
                    },
                },
            };
        } catch (e) {
            if (e.code === 'BotNotInConversationRoster') {
                return {
                    task: {
                        type: 'continue',
                        value: {
                            card: GetJustInTimeCardAttachment(),
                            height: 400,
                            title: 'Adaptive Card - App Installation',
                            width: 300
                        },
                    },
                };
            }
            throw e;
        }
    }
```

Flows a little better if we return from within the `try`.

Also, we try to avoid `var` since [hoisting](https://www.w3schools.com/js/js_hoisting.asp) isn't very intuitive.